File: mobile_flutter/lib/services/auth_service.dart
import 'package:dio/dio.dart';
import 'package:flutter_secure_storage/flutter_secure_storage.dart';
import 'api_service.dart';

class AuthService {
  final ApiService _apiService = ApiService();
  final FlutterSecureStorage _storage = const FlutterSecureStorage();

  Future<Map<String, dynamic>> login(String email, String password) async {
    try {
      final response = await _apiService.dio.post('/auth/login', data: {
        'email': email,
        'password': password,
      });

      if (response.data['access_token'] != null) {
        await _storage.write(key: 'auth_token', value: response.data['access_token']);
        return response.data;
      } else {
        throw Exception('No token returned');
      }
    } on DioException catch (e) {
      throw Exception(e.response?.data['message'] ?? 'Login failed');
    }
  }

  Future<void> register(String email, String password, String firstName, String lastName) async {
    try {
      await _apiService.dio.post('/auth/register', data: {
        'email': email,
        'password': password,
        'firstName': firstName,
        'lastName': lastName,
        'role': 'driver',
      });
    } on DioException catch (e) {
      throw Exception(e.response?.data['message'] ?? 'Registration failed');
    }
  }

  Future<void> logout() async {
    await _storage.delete(key: 'auth_token');
  }

  Future<String?> getToken() async {
    return await _storage.read(key: 'auth_token');
  }
}

File: mobile/services/auth.service.ts
import { api } from './api';
import * as SecureStore from 'expo-secure-store';

export const AuthService = {
    async login(email: string, password: string) {
        try {
            const response = await api.post('/auth/login', { email, password });
            if (response.data.access_token) {
                await SecureStore.setItemAsync('auth_token', response.data.access_token);
                // Store user info if needed, or fetch it separately
                await SecureStore.setItemAsync('user_email', email);
            }
            return response.data;
        } catch (error) {
            throw error;
        }
    },

    async register(data: { firstName: string; lastName: string; email: string; password: string }) {
        try {
            // Backend expects email, password, role.
            // Profile details might need a separate call or backend adjustment.
            // For now, mapping to backend expectation for User model
            const response = await api.post('/auth/register', {
                email: data.email,
                password: data.password,
                role: 'driver', // Mobile app is for drivers
            });

            // Ideally here we would also create the Driver profile entry
            // await api.post('/drivers', { firstName: data.firstName, ... });

            return response.data;
        } catch (error) {
            throw error;
        }
    },

    async logout() {
        await SecureStore.deleteItemAsync('auth_token');
        await SecureStore.deleteItemAsync('user_email');
    },

    async getToken() {
        return await SecureStore.getItemAsync('auth_token');
    }
};
